app/experimenter/nimbus-ui/src/components/SidebarActions/CloneDialog/index.tsx

```diff
@@ -0,0 +1,134 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from "react";
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+import Modal from "react-bootstrap/Modal";
+import { FormProvider } from "react-hook-form";
+import { useCommonForm } from "../../../hooks";
+import { REQUIRED_FIELD } from "../../../lib/constants";
+import { getExperiment_experimentBySlug } from "../../../types/getExperiment";
+import { SlugTextControl } from "../../SlugTextControl";
+
+type CloneDialogProps = {
+  experiment: getExperiment_experimentBySlug;
+  show: boolean;
+  isLoading: boolean;
+  isServerValid: boolean;
+  submitErrors: SerializerMessages;
+  setSubmitErrors: React.Dispatch<React.SetStateAction<Record<string, any>>>;
+  onCancel: () => void;
+  onSave: (params: CloneParams) => void;
+};
+
+export type CloneParams = {
+  name: string;
+};
+
+export const CloneDialog = ({
+  show,
+  experiment,
+  isLoading,
+  isServerValid,
+  submitErrors,
+  setSubmitErrors,
+  onCancel,
+  onSave,
+}: CloneDialogProps) => {
+  const defaultValues: CloneParams = {
+    name: `${experiment!.name} Copy`,
+  };
+
+  const {
+    FormErrors,
+    formControlAttrs,
+    isValid,
+    handleSubmit,
+    formMethods,
+    getValues,
+  } = useCommonForm<keyof CloneParams>(
+    defaultValues,
+    isServerValid,
+    submitErrors,
+    setSubmitErrors,
+  );
+
+  const { trigger } = formMethods;
+
+  const handleSave = handleSubmit(onSave);
+
+  const nameControlAttrs = formControlAttrs("name", {
+    ...REQUIRED_FIELD,
+    maxLength: {
+      value: 80,
+      message: "Cannot be greater than 80 characters",
+    },
+  });
```

Comment:
We have another ticket https://github.com/mozilla/experimenter/issues/5600 open to go in and validate all the different identifiers that make their way out in the DTO, so we'll have to go in and tighten all the validation on these in the serializers, so I'd leave this out of here and just rely on the server validation.